src/main/java/hana/ui/Ui.java

```diff
@@ -0,0 +1,138 @@
+ */
+public class Ui {
+    private static final String NAME = "Hana";
+    private StringBuilder responseBuilder;
+
+    /**
+     * Constructs a new Ui.
+     * Initializes the response builder.
+     */
+    public Ui() {
+        responseBuilder = new StringBuilder();
+    }
+
+    /**
+     * Retrieves the accumulated response and clears the buffer.
+     *
+     * @return The accumulated response.
+     */
+    public String getResponse() {
+        String response = responseBuilder.toString();
+        responseBuilder.setLength(0); // Clear the response builder
+        return response;
+    }
+
+    /**
+     * Prints greetings.
+     */
+    public void printGreetings() {
+        responseBuilder.append(" Hello! I'm ").append(NAME).append(".\n");
+        responseBuilder.append(" What can I do for you?\n");
+    }
+
+    /**
+     * Prints bye message.
+     */
+    public void printBye() {
+        responseBuilder.append("Bye. The application will close in 3 seconds.\n");
+        responseBuilder.append("Hope to see you again soon!\n");
+    }
+
+    /**
+     * Prints error message.
+     */
+    public void printError(String message) {
+        responseBuilder.append("Error: ").append(message).append("\n");
+    }
+
+    /**
+     * Prints message after task successfully added.
+     */
+    public void printAdd(Task task, int taskCount) {
+        responseBuilder.append("Got it. I've added this task:\n");
+        responseBuilder.append("    ").append(task).append("\n");
+        responseBuilder.append("Now you have ").append(taskCount).append(" tasks in the list.\n");
+    }
+
+    /**
+     * Prints message after task successfully marked.
+     */
+    public void printMarked(Task task, boolean isDone) {
+        if (isDone) {
+            responseBuilder.append("Nice! I've marked this task as done:\n");
+        } else {
+            responseBuilder.append("OK, I've marked this task as not done yet:\n");
+        }
+        responseBuilder.append("  ").append(task).append("\n");
+    }
+
+    /**
+     * Prints message after task successfully deleted.
+     */
+    public void printDeleted(Task task, int taskCount) {
+        responseBuilder.append("Noted. I've removed this task:\n");
+        responseBuilder.append("    ").append(task).append("\n");
+        responseBuilder.append("Now you have ").append(taskCount).append(" tasks in the list.\n");
+    }
+
+    /**
+     * Prints all tasks.
+     */
+    public void printTasks(ArrayList<Task> tasks) {
+        for (int i = 0; i < tasks.size(); i++) {
+            responseBuilder.append((i + 1)).append(". ").append(tasks.get(i)).append("\n");
+        }
+    }
+
+    /**
+     * Prints any message.
+     *
+     * @param message Message to print.
+     */
+    public void printMessage(String message) {
+        responseBuilder.append(message).append("\n");
+    }
+
+    /**
+     * Prints all commands.
+     */
+    public void printCommands() {
```

Comment:
Good method to have! To print out the string outputs based on what the user input is! Good idea!!